src/app/component/MyAccountMyOrders/MyAccountMyOrders.component.js

```diff
@@ -13,15 +13,251 @@ import { PureComponent } from 'react';
+        }
+    }
+
+    renderHeading() {
+        return (
+            <h3
+              block="MyAccountMyOrders"
+              elem="Heading"
+            >
+                { __('Order history') }
+            </h3>
+        );
+    }
+
+    renderOrderTable() {
+        const { orderList } = this.props;
+
+        const items = orderList.length
+            ? orderList
+            : Array.from({ length: 7 }, (_, i) => ({ base_order_info: { id: i } }));
+
+        return (
+            <ul
+              block="MyAccountMyOrders"
+              elem="List"
+            >
+                { this.renderHeadingRow() }
+                { items.map(order => this.renderOrderTableRow(order)) }
+            </ul>
+        );
+    }
+
+    renderHeadingRow() {
+        const { isMobileTab } = this.props;
+
+        if (isMobileTab) return null;
+
+        return (
+            <li
+              block="MyAccountMyOrders"
+              elem="Row"
+            >
+                <ul
+                  block="MyAccountMyOrders"
+                  elem="RowItems"
+                  mods={ { type: 'heading' } }
+                >
+                    { Object.entries(this.tableHeadingRow)
+                        .map(([key, value]) => this.renderOrderTableColumn(key, value, 'heading')) }
+                </ul>
+            </li>
+        );
+    }
+
+    renderOrderTableRow(order) {
+        const {
+            base_order_info: { id },
+            base_order_info,
+            shipping_info: {
+                shipping_address: {
+                    is_b2b
+                } = {}
+            } = {}
+        } = order;
+
+        const isRealOrder = Object.keys(base_order_info).length > 1;
+
+        return (
+            <li
+              block="MyAccountMyOrders"
+              elem="Row"
+              key={ id }
+            >
+                <ul
+                  block="MyAccountMyOrders"
+                  elem="RowItems"
+                >
+                    { isRealOrder
+                        ? Object.entries(base_order_info)
+                            .map(([key, value]) => this.processOrderItem(key, value, is_b2b))
+                        : this.shownOrderValues.map((_, i) => this.renderOrderTableColumn(i, null, 'heading')) }
+                    { this.renderShowMoreLink(id, isRealOrder) }
+                </ul>
+            </li>
+        );
+    }
+
+    renderOrderTableColumn(key, value, type, isb2b) {
+        return (
+            <li
+              block="MyAccountMyOrders"
+              elem="Column"
+              mods={ { type: key } }
+              key={ key }
+            >
+                <div
+                  block="MyAccountMyOrders"
+                  elem="ColumnContent"
+                  mods={ { type } }
+                >
+                    <span block="MyAccountMyOrders" elem="MobileColumnHeading">
+                        { value && `${this.tableHeadingRow[key]}: ` }
```

Comment:
Please no conditional rendering, separate into smaller components, or move to separate function.